migrate/migrate.go

```diff
@@ -0,0 +1,127 @@
+package argocd
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"hash/fnv"
+	"os"
+	"strings"
+
+	"github.com/argoproj/argo-cd/server/application"
+	"github.com/argoproj/argo-cd/server/repository"
+	"github.com/argoproj/argo-cd/util"
+	"github.com/argoproj/argo-cd/util/git"
+	"k8s.io/client-go/kubernetes"
+)
+
+// origRepoURLToSecretName hashes repo URL to the secret name using a formula.
+// Part of the original repo name is incorporated for debugging purposes
+func origRepoURLToSecretName(repo string) string {
+	repo = git.NormalizeGitURL(repo)
+	h := fnv.New32a()
+	_, _ = h.Write([]byte(repo))
+	parts := strings.Split(strings.TrimSuffix(repo, ".git"), "/")
+	return fmt.Sprintf("repo-%s-%v", strings.ToLower(parts[len(parts)-1]), h.Sum32())
+}
+
+// repoURLToSecretName hashes repo URL to the secret name using a formula.
+// Part of the original repo name is incorporated for debugging purposes
+func repoURLToSecretName(repo string) string {
+	repo = strings.ToLower(git.NormalizeGitURL(repo))
+	h := fnv.New32a()
+	_, _ = h.Write([]byte(repo))
+	parts := strings.Split(strings.TrimSuffix(repo, ".git"), "/")
+	return fmt.Sprintf("repo-%s-%v", parts[len(parts)-1], h.Sum32())
+}
+
+func renameSecrets(clientOpts argocdclient.ClientOptions) {
+	config, err := clientConfig.ClientConfig()
+	if err != nil {
+		panic(err)
+	}
+
+	namespace, _, err := clientConfig.Namespace()
+	if err != nil {
+		panic(err)
+	}
+
+	kubeclientset := kubernetes.NewForConfigOrDie(config)
+
+	conn, repoIf := argocdclient.NewClientOrDie(clientOpts).NewRepoClientOrDie()
+	defer util.Close(conn)
+	repos, err := repoIf.List(context.Background(), &repository.RepoQuery{})
+	if err != nil {
+		panic(err)
+	}
+
+	for _, repo := range repos {
+		oldSecretName := origRepoURLToSecretName(repo.Name)
+		repoSecret, err = kubeclientset.CoreV1().Secrets(namespace).Get(oldSecretName, metav1.GetOptions{})
+		if err != nil {
+			panic(err)
+		}
+
+		newSecretName := repoURLToSecretName(repo.Name)
+		repoSecret.ObjectMeta.Name = newSecretName
+
+		repoSecret, err = kubeclientset.CoreV1().Secrets(namespace).Create(repoSecret)
+		if err != nil {
+			panic(err)
```

Comment:
same comment here, if it exists, I think we should check that the name is already normalized, if not update the name field.